lib/jekyll/utils.rb

```diff
@@ -297,5 +287,36 @@ def merged_file_read_opts(site, opts)
       merged
     end
 
+    private
+    def merge_values(target, overwrite)
+      target.merge!(overwrite) do |_key, old_val, new_val|
+        if new_val.nil?
+          old_val
+        elsif mergable?(old_val) && mergable?(new_val)
+          deep_merge_hashes(old_val, new_val)
+        else
+          new_val
```

Comment:
I just tried and a lot of tests fail with:
`TypeError: can't dup FalseClass`
looks like we need to check if the value is `duplicable?`.
But it was not duplicating values in the original code here, do we need to do it?